File: src/main/java/com/dude/dms/ui/components/search/TagSearchResult.java
package com.dude.dms.ui.components.search;

import com.dude.dms.backend.data.tags.Tag;
import com.dude.dms.ui.components.dialogs.TagEditDialog;
import com.github.appreciated.card.label.PrimaryLabel;
import com.github.appreciated.card.label.TitleLabel;
import com.vaadin.flow.component.Component;
import com.vaadin.flow.component.orderedlayout.FlexComponent;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;

public class TagSearchResult extends SearchResult {

    private final Tag tag;

    private final long count;

    private final TagEditDialog tagEditDialog;

    public TagSearchResult(Tag tag, long count, TagEditDialog tagEditDialog) {
        this.tag = tag;
        this.count = count;
        this.tagEditDialog = tagEditDialog;
    }

    @Override
    public String getHeader() {
        return "Tag";
    }

    @Override
    public Component getBody() {
        HorizontalLayout h = new HorizontalLayout();
        h.setWidthFull();
        h.add(new TitleLabel(tag.getName()), new PrimaryLabel(count + " Documents"));
        h.setAlignItems(FlexComponent.Alignment.CENTER);
        h.getElement().getStyle().set("border", "5px solid " + tag.getColor());
        return h;
    }

    @Override
    public void onClick() {
        tagEditDialog.open();
    }
}

File: src/main/java/com/dude/dms/ui/components/search/TagSearchResult.kt
package com.dude.dms.ui.components.search

import com.dude.dms.backend.data.Tag
import com.dude.dms.brain.t
import com.dude.dms.ui.components.dialogs.TagEditDialog
import com.github.appreciated.card.label.PrimaryLabel
import com.github.appreciated.card.label.TitleLabel
import com.vaadin.flow.component.orderedlayout.FlexComponent
import com.vaadin.flow.component.orderedlayout.HorizontalLayout

class TagSearchResult(private val tag: Tag, private val count: Long, private val tagEditDialog: TagEditDialog) : SearchResult() {

    override val header = t("tag")

    override val body = HorizontalLayout().apply {
        setWidthFull()
        add(TitleLabel(tag.name), PrimaryLabel("${t("docs")}: $count"))
        alignItems = FlexComponent.Alignment.CENTER
        element.style["border"] = "5px solid " + tag.color
    }

    override fun onClick() {
        tagEditDialog.open()
    }
}